includes/Factory.php

```diff
@@ -19,28 +19,45 @@ class WPNotify_Factory {
 	 */
 	private $recipient_factory;
 
+	/**
+	 * Sender factory implementation to use
+	 *
+	 * @var WPNotify_SenderFactory
+	 */
+	private $sender_factory;
+
 	public function __construct(
 		WPNotify_MessageFactory $message_factory,
-		WPNotify_RecipientFactory $recipient_factory
+		WPNotify_RecipientFactory $recipient_factory,
+		WPNotify_SenderFactory $sender_factory
 	) {
-
 		$this->message_factory   = $message_factory;
 		$this->recipient_factory = $recipient_factory;
+		$this->sender_factory    = $sender_factory;
 	}
 
+	/**
+	 * Create a notification instance
+	 *
+	 * @param $args
+	 *
+	 * @return WPNotify_BaseNotification
+	 */
 	public function create( $args ) {
-		list( $message_args, $recipients_args ) = $this->validate( $args );
 
-		$message = $this->message_factory->create( $message_args );
+		[ $message_args, $recipients_args, $sender_args ] = $this->validate( $args );
```

Comment:
Short array syntax is not in the WordPress coding standards...

We should configure the PHP Code Sniffer to help us with these standards.

Reply:
I've fixed it by using array